### DISCUSSION TITLE
Structured Output

### DISCUSSION THREAD

**Discussion by luki139** [microsoft/semantic-kernel#7082] [Q&A] [answered]
I found it rather surprising that there is kernel function like "ExtractJson" in the core_plugins. That left me wondering whether i am pursuing a wrong approach.

My problem is that my kernel function (from prompt) works best with Chain of Thought or going step by step etc. However this makes extracting YAML from the response much harder. The output occasionally is contaminated with artifacts from examples or input and returns e.g. bolded YAML like this "***key:*** value".

Of course i can just create another kernel function, but is there a better approach?

**Comment by sophialagerkranspandey** [comment]
Is the output necessary in between these steps or are you trying to get the output formatted as Json for the final answer? Thanks!

**Reply by lukaszpancer** [reply]
Only the final answer. I am using a single prompt.

**Comment by sophialagerkranspandey** [comment, marked as answer]
With the scenario mentioned above we recommend you learn more here: https://learn.microsoft.com/en-us/dotnet/api/microsoft.semantickernel.connectors.openai.openaipromptexecutionsettings.responseformat?view=semantic-kernel-dotnet#microsoft-semantickernel-connectors-openai-openaipromptexecutionsettings-responseformat
